### ISSUE TITLE
Editor: Add Stop Button

### ISSUE THREAD

**Issue by bakercp** [olab-io/ofSketch#45]
The stop button needs to call the task manager to cancel the task with the task UUID generated by the run button.  This is less important in desktop development and very important during remote / Raspberry Pi development.

**Comment by bakercp** [comment]
@brannondorsey I just fixed the stop button ... it was pretty simple :)

https://github.com/olab-io/ofSketch/commit/f7f2d999d49e3daeb781b36b548d49f8de5a1f85

The issue now is that when re-compiling via the play button, if the `_currentRunTaskId` is valid, stop-project needs to be stopped.  Currently the easiest way to do this will be to call this on the client side _before_ you set the `_currentRunTaskId` to the new value (see SketchEditor.js around line ...383 / 397).

I'll leave it to you to figure out the best way to keep track of that ...

After we get that fixed and update the changelog, I'd say we should release 0.3.0.

**Comment by brannondorsey** [comment]
hmm... it seems like the stop button doesn't work on linux at [v0.3.0-RC1](https://github.com/olab-io/ofSketch/tree/v0.3.0-RC1).

**Comment by bakercp** [comment]
Darn. Will look at it again timing or tomorrow morning.

**Comment by bakercp** [comment]
I can confirm that this is giving on / off behavior on OSX as well.  Strangely, it seems to depend on which project you are running ... let's try to get this figured out form both back and front ends tomorrow in the olab.

**Comment by bakercp** [comment]
OK.  I'm pretty sure it's working in all cases on both OSX and linux.  A few things had to be fixed:
- Apps that had no output were blocking when I was trying to run `getline` on the standard input, preventing them from closing.  To get around this I started checking the `Poco::Pipe`'s readability using `::select`.  This is not cross platform compatible, but it is POSIX compatible, so as long as we compile ofSketch on Windows with MinGW, we should be OK.
- Process IDs were getting mixed up in some cases when apps were launched with `make run`.  In this case several process IDs are created for the `make` command and later for the `bash` and actual app command.  Only the `make` command PID was being returned during launch, so we successfully killed that process, but the child processes (for us, the app) was hanging in some cases.  This is now fixed by running the app directly with a new `RunTask`.  Thus we are no longer using the `make run` command to run the apps.
